modules/face/samples/sample_face_swapping.cpp

```diff
@@ -0,0 +1,195 @@
+#include "opencv2/face.hpp"
+#include "opencv2/imgcodecs.hpp"
+#include "opencv2/highgui.hpp"
+#include "opencv2/photo.hpp"
+
+using namespace cv;
+using namespace cv::face;
+using namespace std;
+unsigned long minv(unsigned long a,unsigned long b);
+void applyAffineTransform(Mat &warpImage, Mat &src, vector<Point2f> &srcTri, vector<Point2f> &dstTri);
+void calculateDelaunayTriangles(Rect rect, vector<Point2f> &points, vector< vector<int> > &delaunayTri);
+void warpTriangle(Mat &img1, Mat &img2, vector<Point2f> &t1, vector<Point2f> &t2);
+unsigned long minv(unsigned long a,unsigned long b){
+    if(a<b)
+        return a;
+    return b;
+}
+// Apply affine transform calculated using srcTri and dstTri to src
+void applyAffineTransform(Mat &warpImage, Mat &src, vector<Point2f> &srcTri, vector<Point2f> &dstTri)
+{
+    // Given a pair of triangles, find the affine transform.
+    Mat warp_mat = getAffineTransform( srcTri, dstTri );
+    // Apply the Affine Transform just found to the src image
+    warpAffine( src, warpImage, warp_mat, warpImage.size(), INTER_LINEAR, BORDER_REFLECT_101);
+}
+// Calculate Delaunay triangles for set of points
+// Returns the vector of indices of 3 points for each triangle
+void calculateDelaunayTriangles(Rect rect, vector<Point2f> &points, vector< vector<int> > &delaunayTri){
+
+    // Create an instance of Subdiv2D
+    Subdiv2D subdiv(rect);
+    // Insert points into subdiv
+    for( vector<Point2f>::iterator it = points.begin(); it != points.end(); it++)
+        subdiv.insert(*it);
+    vector<Vec6f> triangleList;
+    subdiv.getTriangleList(triangleList);
+    vector<Point2f> pt(3);
+    vector<int> ind(3);
+    for( size_t i = 0; i < triangleList.size(); i++ )
+    {
+        Vec6f triangle = triangleList[i];
+        pt[0] = Point2f(triangle[0], triangle[1]);
+        pt[1] = Point2f(triangle[2], triangle[3]);
+        pt[2] = Point2f(triangle[4], triangle[5]);
+        if ( rect.contains(pt[0]) && rect.contains(pt[1]) && rect.contains(pt[2])){
+            for(int j = 0; j < 3; j++)
+                for(size_t k = 0; k < points.size(); k++)
+                    if(abs(pt[j].x - points[k].x) < 1.0 && abs(pt[j].y - points[k].y) < 1)
+                        ind[j] =(int) k;
+            delaunayTri.push_back(ind);
+        }
+    }
+}
+// Warps and alpha blends triangular regions from img1 and img2 to img
+void warpTriangle(Mat &img1, Mat &img2, vector<Point2f> &t1, vector<Point2f> &t2)
```

Comment:
Sir I tried warping it directly too. The results were less accurate. Using Delaunay transformation helps us to warp only the part that we actually want to swap. Direct warping resulted in warping of some extra part (background) too.